docs/multi-pp-testing/README.md

```diff
@@ -0,0 +1,540 @@
+        --rpc-url $l2_pp2_url \
+        --destination-address $target_address \
+        --private-key $private_key \
+        --bridge-limit 100
+polycli ulxly claim-everything \
+        --bridge-address $bridge_address \
+        --bridge-service-map 1=$l2_pp1b_url \
+        --bridge-service-map 2=$l2_pp2b_url \
+        --bridge-service-map 3=$l2_fepb_url \
+        --rpc-url $l2_fep_url \
+        --destination-address $target_address \
+        --private-key $private_key \
+        --bridge-limit 100
+```
+
+Hopefully at this point everything has been claimed on L2. These
+calls should return empty.
+
+```
+curl -s $l2_pp1b_url/bridges/$target_address | jq '.deposits[] | select(.ready_for_claim) | select(.claim_tx_hash == "")'
+curl -s $l2_pp2b_url/bridges/$target_address | jq '.deposits[] | select(.ready_for_claim) | select(.claim_tx_hash == "")'
+curl -s $l2_fepb_url/bridges/$target_address | jq '.deposits[] | select(.ready_for_claim) | select(.claim_tx_hash == "")'
+```
+
+Let's see if our balances have grown (all should be non-zero)
+
+```
+cast balance --ether --rpc-url $l2_pp1_url $target_address
+cast balance --ether --rpc-url $l2_pp2_url $target_address
+cast balance --ether --rpc-url $l2_fep_url $target_address
+```
+
+Let's check our L2 Pol token balance (all should be non-zero)
+
+```
+token_hash=$(cast keccak $(cast abi-encode --packed 'f(uint32,address)' 0 $pol_address))
+l2_pol_address=$(cast call --rpc-url $l2_pp1_url  $bridge_address 'tokenInfoToWrappedToken(bytes32)(address)' $token_hash)
+cast call --rpc-url $l2_pp1_url $l2_pol_address 'balanceOf(address)(uint256)' $target_address
+cast call --rpc-url $l2_pp2_url $l2_pol_address 'balanceOf(address)(uint256)' $target_address
+cast call --rpc-url $l2_fep_url $l2_pol_address 'balanceOf(address)(uint256)' $target_address
+```
+
+We should also check our gas token balance on all networks. The
+first network should have a balance because it's treated as ordinary
+token. The second network should have nothing because the bridge
+would have been received as a native token.
+
+```
+token_hash=$(cast keccak $(cast abi-encode --packed 'f(uint32,address)' 0 $gas_token_address))
+l2_gas_address=$(cast call --rpc-url $l2_pp1_url  $bridge_address 'tokenInfoToWrappedToken(bytes32)(address)' $token_hash)
+cast call --rpc-url $l2_pp1_url $l2_gas_address 'balanceOf(address)(uint256)' $target_address
+cast call --rpc-url $l2_pp2_url $l2_gas_address 'balanceOf(address)(uint256)' $target_address
+cast call --rpc-url $l2_fep_url $l2_gas_address 'balanceOf(address)(uint256)' $target_address
+```
+
+Similarly, we should confirm that on the second network we have some
+wrapped ETH.
+
+```
+pp2_weth_address=$(cast call --rpc-url $l2_pp2_url $bridge_address 'WETHToken()(address)')
+cast call --rpc-url $l2_pp2_url  $pp2_weth_address 'balanceOf(address)(uint256)' $target_address
+```
+
+## Test Bridges
+
+At this point we have enough funds on L2s to start doing some bridge
```

Comment:
Curious did you try to also withdraw more funds our of one the L2 than what was bridged initially? Did it broke the agglayer?

Reply:
Yes, if you do this it will cause an issue. It should only be possible if there is a premine